massa-client/src/cmds.rs

```diff
@@ -469,6 +469,22 @@ impl Command {
                 let amount = parameters[2].parse::<Amount>()?;
                 let fee = parameters[3].parse::<Amount>()?;
 
+                match amount.checked_add(fee) {
+                    Some(total) => {
+                        if let Ok(addresses_info) = client.public.get_addresses(vec![addr]).await {
+                            if !addresses_info.is_empty()
+                                && addresses_info[0].ledger_info.candidate_ledger_info.balance
+                                    < total
+                            {
+                                println!("Warning: this operation may be rejected due to insuffisant balance");
+                            }
+                        }
+                    }
+                    None => {
+                        bail!("The total amount hit the limit overflow, operation rejected")
```

Comment:
I'm not sure we want to fail here, maybe just display the warning and send the operation.

Reply:
Really? I thought that the node may try to process the same operation and panic anyway. So here or later it doesn't change a lot

Reply:
You point is valid and I do not have a clear answer here :/

I would not assume that the client and the node will fail on the same limits. Nor that the client will always be directly connected to the node. But it make sense to reduce as much as possible the load on the node.

Reply:
We should NEVER panic for things that come from the outside (API or network) ! Otherwise the system can be attacked very easily ! If there is an overflow, just consider the operation invalid.

Reply:
In that case it's only the client that panics

Reply:
still not good :) it should just tell us that the operation is invalid

Reply:
There is a lot of cases where the client can panic. For example if a key is not found in the wallet or at every rpc error. But I can also just print a error/warning.

Whatever, the main question is: should I return here without calling the node? Or should I call the node even if I know that it will fail? I think we can just say `Sorry it's impossible ...` and return directly.

Reply:
>There is a lot of cases where the client can panic. For example if a key is not found in the wallet or at every rpc error. But I can also just print a error/warning.

The client should absolutely NOT panic in any of those cases, just print an error explaining what the user typed wrong.
This needs to be corrected. Can someone create the issues ?

In general, untrusted input validation should be there to prevent crashes.

> should I return here without calling the node? Or should I call the node even if I know that it will fail? I think we can just say Sorry it's impossible ... and return directly.

The node should be called anyways.
The warning message is optional: try to get the info, print the warning if necessary, but if gathering info fails, just ignore it.
In any case, try sending the operation, and if there is an error, print the error explaining what happened.

Reply:
Thank you @damip. I'll rewrite this part and request you a new verification. Check this issue #1850, we'll look at this later, anyway we wanted to verify if we can test scenarios with assert_cmd and it was already in the backlog

Reply:
@AureliaDolo feel free to close this thread if you're agree!